Dockerfile

```diff
@@ -1,17 +1,34 @@
-FROM node:12-alpine
-
+FROM node:12-alpine as base
 WORKDIR /app
-
 COPY package*.json ./
+ENV PORT=8080
 
-RUN npm install
-
+FROM base AS dev
+RUN npm ci
 COPY . .
+CMD ["npm", "run", "dev"]
 
+FROM dev AS test
+COPY . .
+CMD ["npm", "run", "test"]
+
+# EXPOSE 8080
+FROM base AS builder
+RUN npm ci
+COPY src src
+COPY webpack* .
+COPY .eslintrc.json .
+COPY .babelrc .
+COPY tsconfig.json .
```

Comment:
I chose to not copy all; this pretty much just excludes default.conf from the dependencies of the builder image. This way, you can change the nginx stuff without forcing webpack to rebuild the same source code. Super fast.